.github/workflows/ci.yml

```diff
@@ -48,6 +64,91 @@ jobs:
             --private-api-key ${{ secrets.ATLAS_QA_PRIVATE_API_KEY }} \
             --project-id ${{ secrets.ATLAS_QA_PROJECT_ID }}
 
+      - name: Set ${{ matrix.package }}-${{ matrix.platform }} outputs
+        id: set-output
+        run: echo '::set-output name=${{ matrix.package }}-${{ matrix.platform }}-clusterName::${{ steps.deploy-mdb-apps.outputs.clusterName }}'
+
+  print-outputs:
+    name: Print env cluster names
+    runs-on: ubuntu-latest
+    needs:
+      - baas_matrix
+    steps:
+      - name: baas-outputs 
+        env:
+          NEEDS: ${{ toJSON(needs) }}
+        run: echo "$NEEDS"
+
+  cleanup-matrix:
+    strategy:
+      matrix:
+        package: [flutter]
+        platform: [linux, windows, macos]
+        include:
+          - platform: linux
+            package: dart
+          - platform: windows
+            package: dart
+          - platform: macos
+            package: dart
+    runs-on: ubuntu-latest
+    name: Cleanup ${{ matrix.platform }} for ${{ matrix.package }}
+    needs:
+      - tests-linux
+      - flutter-linux
+      - tests-macos
+      - flutter-macos
+      - tests-windows
+      - flutter-windows
+      - flutter-ios
```

Comment:
Should we wait for iOS here?

Reply:
I will leave it but I will delete the separated cleanup job for ios. Only Android is too slow.